kv: TestRangeLookupWithOpenTransaction is flaky

The following test appears to have failed:

[#9955](https://circleci.com/gh/cockroachdb/cockroach/9955):

```
W1208 22:12:51.892526 602 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
W1208 22:12:52.135511 602 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
I1208 22:12:52.135659 602 client/db.go:483  failed batch: failed to send RPC request Server.Batch: client is unhealthy
W1208 22:12:52.137379 602 rpc/client.go:249  unexpected EOF
I1208 22:12:52.137626 602 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestAuthentication (0.32s)
	db_test.go:258: failed to send RPC request Server.Batch: client is unhealthy
=== RUN   TestRangeLookupWithOpenTransaction
I1208 22:12:52.183403 602 rpc/clock_offset.go:154  monitoring cluster offset
I1208 22:12:52.204593 602 multiraft/multiraft.go:579  node 1 starting
I1208 22:12:52.205423 602 raft/raft.go:441  [group 1] 1 became follower at term 5
I1208 22:12:52.205664 602 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:12:52.205837 602 multiraft/multiraft.go:999  node 1 campaigning because initial confstate is [1]
I1208 22:12:52.205966 602 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1208 22:12:52.206080 602 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1208 22:12:52.206189 602 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6
--
I1208 22:12:57.185056 602 kv/txn_coord_sender.go:281  txn coordinator: 0.00 txn/sec, 0.00/0.00/0.00 %cmmt/abrt/abnd, 0/0/0 avg/σ/max duration, 0.0/0.0/0.0 avg/σ/max restarts (0 samples)
I1208 22:12:57.222550 602 multiraft/multiraft.go:709  node 1: g is nil for statusOp for gropuID 1
W1208 22:12:57.223200 602 storage/raft_log_queue.go:111  storage/raft_log_queue.go:74: the raft group doesn't exist for range 1
I1208 22:12:57.247870 602 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
==================
WARNING: DATA RACE
Write by goroutine 181:
  testing.tRunner()
      /tmp/workdir/go/src/testing/testing.go:457 +0xef

Previous write by goroutine 53:
  testing.(*common).FailNow()
      /tmp/workdir/go/src/testing/testing.go:335 +0x41
  testing.(*common).Fatal()
      /tmp/workdir/go/src/testing/testing.go:373 +0x7c
  github.com/cockroachdb/cockroach/kv_test.TestRangeLookupWithOpenTransaction.func1()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender_server_test.go:74 +0x18c

Goroutine 181 (running) created at:
  testing.RunTests()
      /tmp/workdir/go/src/testing/testing.go:561 +0xaa3
--
  github.com/cockroachdb/cockroach/kv_test.TestRangeLookupWithOpenTransaction()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender_server_test.go:77 +0x767
  testing.tRunner()
      /tmp/workdir/go/src/testing/testing.go:456 +0xdc
==================
--- FAIL: TestRangeLookupWithOpenTransaction (5.11s)
	dist_sender_server_test.go:74: failed to send RPC request Server.Batch: client is unhealthy
	dist_sender_server_test.go:83: get request did not succeed in face of range metadata intent
=== RUN   TestMultiRangeBatchBounded
I1208 22:12:57.300173 602 rpc/clock_offset.go:154  monitoring cluster offset
I1208 22:12:57.304288 602 multiraft/multiraft.go:579  node 1 starting
I1208 22:12:57.305185 602 raft/raft.go:441  [group 1] 1 became follower at term 5
I1208 22:12:57.305386 602 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:12:57.305522 602 multiraft/multiraft.go:999  node 1 campaigning because initial confstate is [1]
I1208 22:12:57.305624 602 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1208 22:12:57.305718 602 raft/raft.go:454  [group 1] 1 became candidate at term 6
--

I1208 22:13:05.098118 602 stopper.go:241  draining; tasks left:
1      storage/replica.go:1441
I1208 22:13:05.102326 602 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- PASS: TestPropagateTxnOnPushError (0.46s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	50.631s
=== RUN   TestHeartbeatSingleGroup
I1208 22:12:08.039299 577 multiraft/multiraft.go:579  node 1 starting
I1208 22:12:08.040155 577 raft/raft.go:441  [group 1] 1 became follower at term 5
I1208 22:12:08.040385 577 raft/raft.go:234  [group 1] newRaft 1 [peers: [1,2], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:12:08.040614 577 multiraft/multiraft.go:579  node 2 starting
I1208 22:12:08.041292 577 raft/raft.go:441  [group 1] 2 became follower at term 5
I1208 22:12:08.041453 577 raft/raft.go:234  [group 1] newRaft 2 [peers: [1,2], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:12:08.041738 577 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1208 22:12:08.041957 577 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1208 22:12:08.042125 577 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6

```
Please assign, take a look and update the issue accordingly.

--- a/kv/dist_sender_server_test.go
+++ b/kv/dist_sender_server_test.go
@@ -68,16 +68,17 @@ func TestRangeLookupWithOpenTransaction(t *testing.T) {
 	// intent error. If it did, it would go into a deadloop attempting
 	// to push the transaction, which in turn requires another range
 	// lookup, etc, ad nauseam.
-	success := make(chan struct{})
+	errors := make(chan error)
 	go func() {
-		if _, err := db.Get("a"); err != nil {
-			t.Fatal(err)
-		}
-		close(success)
+		_, err := db.Get("a")
+		errors <- err
 	}()
 
 	select {
-	case <-success:
+	case err := <-errors:
+		if err != nil {
+			t.Fatal(err)
+		}
 		// Hurrah!
 	case <-time.After(5 * time.Second):
 		t.Errorf("get request did not succeed in face of range metadata intent")

--- a/kv/dist_sender_server_test.go
+++ b/kv/dist_sender_server_test.go
@@ -68,20 +68,8 @@ func TestRangeLookupWithOpenTransaction(t *testing.T) {
 	// intent error. If it did, it would go into a deadloop attempting
 	// to push the transaction, which in turn requires another range
 	// lookup, etc, ad nauseam.
-	errors := make(chan error)
-	go func() {
-		_, err := db.Get("a")
-		errors <- err
-	}()
-
-	select {
-	case err := <-errors:
-		if err != nil {
-			t.Fatal(err)
-		}
-		// Hurrah!
-	case <-time.After(5 * time.Second):
-		t.Errorf("get request did not succeed in face of range metadata intent")
+	if _, err := db.Get("a"); err != nil {
+		t.Fatal(err)
 	}
 }
 

